cider/exec/nextgen/jitlib/base/JITValue.h

```diff
@@ -127,11 +99,59 @@ class JITValue : public JITBaseValue {
   virtual JITValuePointer dereference() = 0;
 
  private:
+  JITTypeTag type_tag_;
   std::string value_name_;
   JITFunction& parent_function_;
   JITTypeTag sub_type_tag_;
 };
 
+class JITValuePointer {
+ public:
+  JITValuePointer(std::nullptr_t ptr_) : ptr_(ptr_) {}
+
+  JITValuePointer(JITValue* value) : ptr_(value) { ptr_->addRef(); }
```

Comment:
Maybe hide this constructor as protect or private is better, and declare ```makeJITValuePointer``` as a friend. So that ```JITValuePointer``` could be generated from ```makeJITValuePointer``` only.